controllers/nstemplateset/nstemplatetier.go

```diff
@@ -130,15 +132,34 @@ func (t *tierTemplate) convertParametersToMap(runtimeParam map[string]string) ma
 
 }
 
+// processGoTemplates processes the Go templates
 func (t *tierTemplate) processGoTemplates(runtimeParams map[string]string, filters ...template.FilterFunc) ([]runtimeclient.Object, error) {
-	objList := make([]runtimeclient.Object, 0, len(t.ttr.Spec.TemplateObjects))
-	rawExtObjList := make([]runtime.RawExtension, 0, len(t.ttr.Spec.TemplateObjects))
 	paramMap := t.convertParametersToMap(runtimeParams) // go execute requires parameters in form of map
+	var templatesToProcess []runtime.RawExtension
```

Comment:
let's change it to 
```suggestion
	var objectsToProcess []runtime.RawExtension
```
because it's the raw content of the objects that we need to process. Sure, it's part of the whole template, but there is only one template, not multiple of them.
You can also change the name of the function to the singular form `processGoTemplate`